app/controllers/admin/claims_controller.rb

```diff
@@ -22,14 +22,16 @@ def show
   end
 
   def search
-    return unless params[:reference].present?
+    return unless params[:query].present?
 
-    claim = Claim.find_by(reference: params[:reference].upcase)
+    @claims = Claim::Search.new(params[:query]).claims.includes(:eligibility)
 
-    if claim
-      redirect_to(admin_claim_url(claim))
+    if @claims.blank?
+      flash.now[:notice] = "Cannot find a claim for query \"#{params[:query]}\""
+    elsif @claims.count == 1
+      redirect_to(admin_claim_url(@claims.first))
     else
```

Comment:
Reckon we can get rid of this last `else` as it's redundant: if the redirect doesn't happen above, the controller will render the `search` template anyway.

Reply:
Yeah, I just thought it might be a bit clearer to spell it out, but, on reflection, that statement does look a bit icky, and as it's default behaviour, shouldn't be too surprising